mobile/src/main/java/org/openhab/habdroid/model/ServerProperties.java

```diff
@@ -119,6 +124,19 @@ public void onSuccess(String response, Headers headers) {
                     } catch (NumberFormatException nfe) {
                         // ignored: older versions without SSE support didn't return a number
                     }
+
+                    try {
+                        JSONArray jsonArray = result.getJSONArray("links");
+                        for (int i = 0; i < jsonArray.length(); i++) {
+                            JSONObject extensionJson = jsonArray.getJSONObject(i);
+                            if (extensionJson.getString("type").equals("habpanel")) {
+                                flags |= SERVER_FLAG_HABPANEL_INSTALLED;
+                            }
+                        }
+                    } catch (JSONException e) {
```

Comment:
This feels a bit ... unclean to me. Why not use `optJSONArray()` and check for null?

Similarly for the if check, just do `if ("habpanel".equals(extensionJson.optString("type"))`

Reply:
Done